src/utils/saml.ts

```diff
@@ -68,7 +121,7 @@ export const getPreValidateResponse = (
   doneCb,
   callback
   // eslint-disable-next-line sonarjs/cognitive-complexity
-): Promise<void | E.Left<void> | E.Right<void>> => {
+): Promise<void | E.Either<void, void>> => {
```

Comment:
It's out of scope for this PR, but `Promise<Either<L,R>>` isn't `TaskEither<L,R>`?

Also, what `Promise<void>` stands for?

Reply:
you are right, it is a `TaskEither<void,void>` in this case. should I do this change in another PR? it was a small catch I saw while refactoring

Reply:
There is a small difference from `Promise<Either<L,R>>` and `TaskEither<L,R>`, the task either operation *can't fail* instead the Promise can do. The conversion can be made catching the promise error and putting it on the left `TaskEither<E|L, R>`.

Reply:
You are right, indeed `TaskEither<L,R> = () => Promise<Either<L,R>>`

Dumb request I made, sorry